### PR TITLE
IBX-186: The top button of side menus gets hidden at the bottom of the page

### PR DESCRIPTION
**Pull request by GrabowskiM**
| Question      | Answer
| ------------- | ---
| Tickets       | IBX-186
| Bug fix?      | yes
| New feature?  | no
| BC breaks?    | no
| Tests pass?   | yes
| Doc needed?   | no
| License       | [GPL-2.0](https://github.com/ezsystems/ezplatform-admin-ui/blob/master/LICENSE)

#### Checklist:
- [x] Coding standards (`$ composer fix-cs`)
- [x] Ready for Code Review

### CODE REVIEW

File: src/bundle/Resources/public/js/scripts/sidebar/extra.actions.js [hunk 1]
```diff
@@ -3,9 +3,33 @@
...
+    let resizeAndScrollTimeout;
     const haveHiddenPart = (element) => element.classList.contains(CLASS_HIDDEN) && !element.classList.contains(CLASS_PREVENT_SHOW);
+    const setContainerHeight = () => {
+        const container = doc.querySelector('.ez-extra-actions:not(.ez-extra-actions--hidden)');
+        const bottomPosition = Math.min(footer.getBoundingClientRect().top, global.innerHeight);
+        const containerHeight = bottomPosition - container.getBoundingClientRect().top - ACTIONS_CONTAINER_MARGIN;
+
+        container.style.height = `${containerHeight}px`;
+    };
+    const setHeightOnScrollAndResize = () => {
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/sidebar/extra.actions.js:20]:
```suggestion
    const setContainerHeightTimeout = () => {
```
?

File: src/bundle/Resources/public/js/scripts/sidebar/extra.actions.js [hunk 2]
```diff
@@ -3,9 +3,33 @@
...
+        clearTimeout(resizeAndScrollTimeout);
+
+        resizeAndScrollTimeout = setTimeout(setContainerHeight, RESIZE_AND_SCROLL_TIMEOUT);
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/sidebar/extra.actions.js:23]:
```suggestion
        setContainerHeightTimeout = setTimeout(setContainerHeight, RESIZE_AND_SCROLL_TIMEOUT);
```
?

File: src/bundle/Resources/public/js/scripts/sidebar/extra.actions.js [hunk 3]
```diff
@@ -3,9 +3,33 @@
...
+    };
+    const setHeightOnResizeAndScroll = () => {
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/sidebar/extra.actions.js:25]:
```suggestion
    const addListeners = () => {
```
?

File: src/bundle/Resources/public/js/scripts/sidebar/extra.actions.js [hunk 4]
```diff
@@ -3,9 +3,33 @@
...
+        global.addEventListener('scroll', setHeightOnScrollAndResize, false);
+        global.addEventListener('resize', setHeightOnScrollAndResize, false);
+    };
+    const resetHeightOnResizeAndScroll = () => {
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/sidebar/extra.actions.js:29]:
```suggestion
    const removeListeners = () => {
```
?

**Review by dew326 — CHANGES_REQUESTED** [review]
I think we can do better if it comes to naming. I suggested some names, feel free to enhance them.

File: src/bundle/Resources/public/js/scripts/sidebar/extra.actions.js [hunk 5]
```diff
@@ -33,6 +57,7 @@
                         }
 
                         doc.body.removeEventListener('click', detectClickOutside, false);
+                        resetHeightOnResizeAndScroll();
```
Comment (dew326) [inline src/bundle/Resources/public/js/scripts/sidebar/extra.actions.js:60]: Error function not found?

**Review by tischsoic — APPROVED** [review]

**Review by dew326 — APPROVED** [review]

**Review by lucasOsti — APPROVED** [review]

**Comment by lserwatka** [comment]
You can merge it up.

**Pull request merged**